Content.Shared/Crawling/CrawlingSystem.cs

```diff
@@ -0,0 +1,130 @@
+        SubscribeLocalEvent<CrawlingComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshMovespeed);
+
+        CommandBinds.Builder
+            .Bind(ContentKeyFunctions.ToggleCrawling, InputCmdHandler.FromDelegate(ToggleCrawlingKeybind, handle: false))
+            .Register<CrawlingSystem>();
+    }
+
+    private void ToggleCrawlingKeybind(ICommonSession? session)
+    {
+        if (session?.AttachedEntity == null)
+            return;
+        var ev = new CrawlingKeybindEvent();
+        RaiseLocalEvent(session.AttachedEntity.Value, ev);
+    }
+    private void ToggleCrawling(EntityUid uid, CrawlerComponent component, CrawlingKeybindEvent args)
+    {
+        ///checks players standing state, downing player if they are standding and starts doafter with standing up if they are downed
+        switch (_standing.IsDown(uid))
+        {
+            case false:
+                _standing.Down(uid, dropHeldItems: false);
+                break;
+            case true:
+                _doAfter.TryStartDoAfter(new DoAfterArgs(EntityManager, uid, component.StandUpTime, new CrawlStandupDoAfterEvent(),
+                uid, used: uid)
+                {
+                    BreakOnDamage = true
+                });
+                break;
+        }
+    }
+    private void OnCrawlingAlertEvent(EntityUid uid, CrawlerComponent component, CrawlingAlertEvent args)
+    {
+        var ev = new CrawlingKeybindEvent();
+        RaiseLocalEvent(args.User, ev);
+    }
+    private void OnDoAfter(EntityUid uid, CrawlerComponent component, CrawlStandupDoAfterEvent args)
+    {
+        if (args.Cancelled)
+            return;
+        _standing.Stand(uid);
+    }
+    private void OnStandUp(EntityUid uid, CrawlerComponent component, StandAttemptEvent args)
+    {
+        if (args.Cancelled)
+            return;
+        RemCompDeferred<CrawlingComponent>(uid);
+        _alerts.ClearAlert(uid, component.CtawlingAlert);
+    }
+    private void OnFall(EntityUid uid, CrawlerComponent component, DownAttemptEvent args)
+    {
+        if (args.Cancelled)
+            return;
+        _alerts.ShowAlert(uid, component.CtawlingAlert);
+        if (!HasComp<CrawlingComponent>(uid))
+            AddComp<CrawlingComponent>(uid);
+        //TODO: add hiding under table
+    }
+    private void OnStunned(EntityUid uid, CrawlerComponent component, StunnedEvent args)
+    {
+        if (!HasComp<CrawlingComponent>(uid))
+            AddComp<CrawlingComponent>(uid);
+        _alerts.ShowAlert(uid, component.CtawlingAlert);
+    }
+    private void OnGetExplosionResistance(EntityUid uid, CrawlerComponent component, ref GetExplosionResistanceEvent args)
+    {
+        // fall on explosion damage and lower explosion damage of crawling
+        if (_standing.IsDown(uid))
+            args.DamageCoefficient *= component.DownedDamageCoefficient;
+        else
+            _standing.Down(uid, dropHeldItems: false);
```

Comment:
```suggestion
            _standing.Down(uid);
```

I think, dropHeldItems should be true. This is explosion!
`_standing.Down(uid);` should be moved to `BeforeExplodeEvent`